sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/training/CustomModelTransforms.java

```diff
@@ -67,42 +71,139 @@ static CustomFormModel toCustomFormModel(Model modelResponse) {
+            } else {
+                formType = formType + composedTrainResultItem.getModelId().toString();
+            }
+            Map<String, CustomFormModelField> fieldMap = new TreeMap<>();
+            composedTrainResultItem.getFields()
+                .forEach(formFieldsReport -> fieldMap.put(
+                    formFieldsReport.getFieldName(),
+                    new CustomFormModelField(
+                        null,
+                        formFieldsReport.getFieldName(),
+                        formFieldsReport.getAccuracy())));
+
+            CustomFormSubmodel customFormSubmodel =
+                new CustomFormSubmodel(
+                    composedTrainResultItem.getAverageModelAccuracy(),
+                    fieldMap,
+                    formType);
+            PrivateFieldAccessHelper.set(customFormSubmodel, "modelId",
+                composedTrainResultItem.getModelId().toString());
+            subModelList.add(customFormSubmodel);
+        }
+        return subModelList;
+    }
+
+    /** Creates a submodel list from labeled models service data **/
+    private static List<CustomFormSubmodel> getLabeledSubmodels(Model modelResponse, String modelId, String formType) {
+        Map<String, CustomFormModelField> fieldMap = new TreeMap<>();
+        List<CustomFormSubmodel> subModelList = new ArrayList<>();
+        modelResponse.getTrainResult().getFields()
+            .forEach(formFieldsReport -> fieldMap.put(formFieldsReport.getFieldName(),
+                new CustomFormModelField(null,
+                    formFieldsReport.getFieldName(),
+                    formFieldsReport.getAccuracy())));
+
+        CustomFormSubmodel customFormSubmodel =
+            new CustomFormSubmodel(
+                modelResponse.getTrainResult().getAverageModelAccuracy(),
+                fieldMap,
+                formType);
+        PrivateFieldAccessHelper.set(customFormSubmodel, "modelId", modelId);
+        subModelList.add(customFormSubmodel);
+        return subModelList;
+    }
+
+    /** Creates a submodel list from unlabeled models service data **/
+    private static List<CustomFormSubmodel> getUnlabeledSubmodels(Map<String, List<String>> modelResponseClusters,
+        String modelId, String formType) {
+        List<CustomFormSubmodel> subModelList = new ArrayList<>();
+        modelResponseClusters
+            .forEach((clusterKey, clusterFields) -> {
+                Map<String, CustomFormModelField> fieldMap = new TreeMap<>();
+                forEachWithIndex(clusterFields, (index, eachField) -> {
+                    String fieldName = "field-" + index;
+                    fieldMap.put(fieldName, new CustomFormModelField(eachField, fieldName, null));
+                });
+                CustomFormSubmodel customFormSubmodel = new CustomFormSubmodel(
+                    null,
+                    fieldMap,
+                    formType + clusterKey);
```

Comment:
Is this setting formType to "custom:<cluster-id>? I think we are leaving unsupervised untouched so it should remain "form-<cluster-id>.

Unsupervised doesn't return docType or docTypeConfidence so this field was something we created/set in the SDK. I don't see a reason to alter it from what we returned in the GA version of the library.